optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -143,14 +148,43 @@ public void solve(DefaultSolverScope<Solution_> solverScope) {
                 stepEnded(stepScope);
                 phaseScope.setLastCompletedStepScope(stepScope);
             }
-        } finally {
+        } catch (RuntimeException e) {
             // If a partition thread throws an Exception, it is propagated here
             // but the other partition thread won't finish any time soon, so we need to terminate them
             childThreadPlumbingTermination.terminateChildren();
+        } finally {
```

Comment:
Based on the comment I understand that it's only needed to manually terminate children if one of them threw exception. If that doesn't happen it should be guaranteed that all children have terminated simply by finishing iteration over the PartitionQueue. Terminating children in finally had one not-so-nice side-effect of printing the _"Terminating child solvers early."_ even if that's not the case.